specification/communication/data-plane/Email/preview/2024-07-01-preview/CommunicationServicesEmail.json

```diff
@@ -0,0 +1,390 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "EmailClient",
+    "description": "Azure Communication Email Service",
+    "version": "2024-07-01-preview"
+  },
+  "paths": {
+    "/emails/operations/{operationId}": {
+      "get": {
+        "tags": [
+          "EmailGetSendResult"
+        ],
+        "summary": "Gets the status of the email send operation.",
+        "operationId": "Email_GetSendResult",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "operationId",
+            "description": "ID of the long running operation (GUID) returned from a previous call to send email",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "headers": {
+              "x-ms-error-code": {
+                "description": "Error code - this will be the same as the code in the error property in the response body.",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "../../../../../common-types/data-plane/v1/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Message status was successfully retrieved.",
+            "headers": {
+              "retry-after": {
+                "description": "This header will only be present when the status is a non-terminal status. It indicates the minimum amount of time in seconds to wait before polling for operation status again.",
+                "type": "integer",
```

Comment:
Are you sure this should be `integer` instead of `string`?  The HTTP spec seems to require `string` to support date-based values.

https://www.rfc-editor.org/rfc/rfc9110#field.retry-after

Reply:
I can check the historical reasoning of this choice with the team and get back to you. That being said, I have a couple of follow up questions here: 

If it's the spec that is incorrect, I'm curious as to why these errors are only manifesting in our builds now, despite using the same version of autorest? It appears that it wasn't caught in our team's previous PRs. Examples: [Add base64 formatting to contentInBase64 property by kagbakpem · Pull Request #24617 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/24617), [Azure Communication Services Email API for GA by apattath · Pull Request #22979 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/22979).
 
If I make changes to the older specs retroactively, are there any considerations here with introducing breaking changes? It seems like in our most recent PR, a tag was added to approve a breaking change (unsure if it was for the same reason). Is that appropriate here?

Reply:
For breaking changes, we regard the service behavior as the "source of truth". If the API definition changes to more accurately describe the service behavior, we allow this and actually encourage it.

Reply:
The check also failed in PR 22979.  For PR 24617, the check was bypassed using label `Approved-BreakingChange`, and we no longer have the build log, but I suspect it also failed with the same error.

I think the first question here, is how should `retry-after` be expressed in services, specs, and examples?  Always a `string`, or can each service decide between `string` and `integer`?

For instance, even though the HTTP spec allows string dates, if a service has decided it will **ever** only send integer seconds in the header, would it be appropriate for the service and spec to use `integer`?

@mikekistler: Do you know the answer?

Reply:
We always want the value of `retry-after` to be "delay-seconds" and not a date, and RFC 7231 says:
> A delay-seconds value is a non-negative decimal integer, representing time in seconds.

So `integer` is the appropriate type for this header value.

Furthermore, it is common that string and integer values are serialized in headers in the same way, so it might be possible to change the API description from `type: string` to `type: integer` in a non-breaking way (i.e. without changing the behavior of the service). As long as the service does not wrap the value in quotes, e.g.

```
Retry-after: "180"
```

but most services will not do this so it might be fine.

Reply:
>We always want the value of retry-after to be "delay-seconds" and not a date

@mikekistler: Can you elaborate on this?  Is this an Azure guideline above the HTTP spec (which allows either)?  Why wouldn't we want to give services the option to send either seconds or a date?

Reply:
`Retry-after` should always specific `delay-seconds` in any response.